src/View/Helper/FormHelper.php

```diff
@@ -1433,6 +1433,13 @@ protected function _magicOptions($fieldName, $options, $allowOverride)
             $options['required'] = $context->isRequired($fieldName);
         }
 
+        $message = $context->getErrorMessage($fieldName);
+        if ($options['required'] && $message) {
+            $message = htmlspecialchars(addslashes($message));
```

Comment:
Should we use `h()` to handle application specific encodings?

Reply:
Makes sense. Doing so will require `cakephp/core` in the `cakephp/validation` package.

Reply:
Why? This message is being escaped in View/Helper/FormHelper.

Reply:
Because I'm dumb and got confused about which file I was looking at :D